src/HealthChecks.AzureIoTHub/AzureIoTHubtHealthCheck.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices;
+
+namespace HealthChecks.AzureIoTHub
+{
+    public class AzureIoTHubHealthCheck : IHealthCheck
+    {
+        private readonly AzureIoTHubOptions _options;
+        public AzureIoTHubHealthCheck(AzureIoTHubOptions options)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                if (_options.RegistryWriteCheck)
+                {
+                    using (var client = RegistryManager.CreateFromConnectionString(_options.ConnectionString))
+                    {
+                        var deviceId = _options.RegistryWriteDeviceIdFactory();
+
+                        await client.AddDeviceAsync(new Device(deviceId), cancellationToken);
```

Comment:
What if device with that id already exists?

Reply:
Exception will throw, I think, and unhealthy result will be. I'll check this scenario.

Reply:
Reimplemented:
if device with id already exists -- remove it
if device with id not exists -- create and remove it